src/algebra/is_prime_pow.lean

```diff
@@ -133,6 +132,23 @@ begin
   simpa using hm₁,
 end
 
+lemma is_prime_pow_iff_factorization_single (n : ℕ):
+  is_prime_pow n ↔ ∃ p k : ℕ, 0 < k ∧ n.factorization = finsupp.single p k :=
+begin
+  rw is_prime_pow_nat_iff,
+  split,
+  { rintros ⟨p, k, pp, hk, hn⟩, use [p, k, hk],
+    rw [←hn, nat.prime.factorization_pow pp] },
+  { rintros ⟨p, k, hk, hn⟩,
+    rcases eq_or_ne n 0 with rfl | hn0,
+    { apply absurd _ (@irrefl ℕ (<) _ 0), convert hk, simpa [eq_comm] using hn },
+    { rw nat.pow_of_factorization_single hn0 hn,
+      refine ⟨p, k, _, hk, rfl⟩,
+      apply nat.prime_of_mem_factorization,
+      rw hn,
+      simp [hk.ne'] } }
```

Comment:
I think doing a case split when one case can't happen is a little confusing. How about:
```suggestion
    have hn0 : n ≠ 0,
    { rintro rfl,
      simp only [finsupp.single_eq_zero, eq_comm, nat.factorization_zero] at hn,
      exact hk.ne' hn },
    rw nat.pow_of_factorization_single hn0 hn,
    refine ⟨p, k, _, hk, rfl⟩,
    apply nat.prime_of_mem_factorization,
    rw hn,
    simp [hk.ne'] }
```

Reply:
Yes, that's a nicer way to write it, thanks